compiler/parser/parser.peg

```diff
@@ -758,12 +747,9 @@ Primary
   / "(" __ expr:Expr __ ")" { RETURN(expr) }
 
 OverExpr
-  = "over" _ exprs:Exprs _ "with" _ locals:LetAssignments __ "|" __ scope:Sequential {
+  = "over" _ exprs:Exprs locals:(_ "with" _ l:LetAssignments { RETURN(l) })? __ "|" __ scope:Sequential {
```

Comment:
Idk about having a RETURN in the middle of the definition here, maybe break the `with` into its own statement?